### ISSUE TITLE
Explicit interpreters instead of extension methods

### ISSUE THREAD

**Issue by adamw** [softwaremill/tapir#889]
Right now, client/server/doc interpreters use extension methods which are imported from package objects, e.g.:

```scala
import sttp.tapir.server.akkahttp._
// toRoute is an extension method
myEndpoint.toRoute(logic) 
// or
myEndpoint.serverLogic(logic).toRoute
// or
List(
  myEndpoint1.serverLogic(logic1),
  myEndpoint2.serverLogic(logic2)
).toRoute
``` 

While extension methods are good for adding functionalities which somehow extend, change, convert format or summarize the given data structure, I'm not sure if they are the best tool here. We're interpreting one data structure as something completely different, like an akka-http route or OpenAPI docs. Considerable amount of code is involved. It might not be that obvious what is going on here, and maybe the following would be more readable and easier to import:

```scala
import sttp.tapir.server.akkahttp.AkkaHttpInterpreter
// toRoute is an extension method
AkkaHttpInterpreter.toRoute(myEndpoint)(logic) 
// or
AkkaHttpInterpreter.toRoute(myEndpoint.serverLogic(logic))
// or
AkkaHttpInterpreter.toRoute(List(
  myEndpoint1.serverLogic(logic1),
  myEndpoint2.serverLogic(logic2)
))
```

Similarly, we would have `OpenAPIInterpreter`, `Http4sInterpreter`, `SttpClientInterpreter` etc. Exisiting extension methods would stay there for some time of course under a deprecation notice.

Good idea / bad idea / comments? :)

**Comment by kubukoz** [comment]
I like it ;) I personally find it easier to find a conversion that's "static" and not added in an implicit class. Besides, these are (at least in my projects) used very, very rarely so the convenience of dot syntax isn't as necessary as it would for usual endpoint->endpoint methods.

**Comment by gaeljw** [comment]
Absolutely agree, and I've already had cases of teammates starting to learn Tapir and having a hard time understand what to import to get the right extension method.

**Comment by lmserrano** [comment]
I agree with this. It makes it easier to follow what is happening and easier for anyone new to Tapir to onboard a project.

**Comment by kubukoz** [comment]
btw. another datapoint, when migrating in https://github.com/pitgull/pitgull/pull/102 I had to spend a moment figuring out why the extension was not working anymore - the types of the receiver have been made more specific, so I just got a compilation error of "method not found" instead of "invalid type parameters" etc. - this can be alleviated by either replacing the type parameters with constraints on the extension methods (`(implicit isStreaming: S <: EntityBody[F])`) or using explicit interpreters.
